poll/poll.py

```diff
@@ -140,53 +142,62 @@ def gather_items(self, data, result, noun, field, image=True):
             source_items = []
             result['success'] = False
             result['errors'].append(
-                "'{0}' is not present, or not a JSON array.".format(field))
+                self.ugettext("'{0}' is not present, or not a JSON array.").format(field))
         else:
             source_items = data[field]
 
         # Make sure all components are present and clean them.
         for item in source_items:
             if not isinstance(item, dict):
                 result['success'] = False
-                result['errors'].append(
-                    "{0} {1} not a javascript object!".format(noun, item))
+                # Translators: {noun} is either "Answer" or "Question". {item} identifies the answer or question.
+                error_message = self.ugettext("{noun} {item} not a javascript object!").format(noun=noun, item=item)
+                result['errors'].append(error_message)
                 continue
             key = item.get('key', '').strip()
             if not key:
                 result['success'] = False
-                result['errors'].append(
-                    "{0} {1} contains no key.".format(noun, item))
+                # Translators: {noun} is either "Answer" or "Question". {item} identifies the answer or question.
+                error_message = self.ugettext("{noun} {item} contains no key.").format(noun=noun, item=item)
+                result['errors'].append(error_message)
             image_link = item.get('img', '').strip()
             image_alt = item.get('img_alt', '').strip()
             label = item.get('label', '').strip()
             if not label:
                 if image and not image_link:
                     result['success'] = False
-                    result['errors'].append(
-                        "{0} has no text or img. Please make sure all {1}s "
-                        "have one or the other, or both.".format(noun, noun.lower()))
+                    # Translators: {noun} is either "Answer" or "Question". {noun_lower} is the lowercase version
+                    # of {noun}.
+                    error_message = self.ugettext("{noun} has no text or img. Please make sure all "
+                                                  "{noun_lower}s have one or the other, or both.").\
+                        format(noun=noun, noun_lower=noun.lower())
```

Comment:
@asadiqbal08 It would be good to stick to edX's formatting standards here, see ["Breaking Long Lines"](https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/style_guides/python_guidelines.html#breaking-long-lines). That will require moving the translator comment inside the `ugettext` call:

```python
                    result['success'] = False
                    error_message = self.ugettext(
                        # Translators: {noun} is either "Answer" or "Question".
                        # {noun_lower} is the lowercase version of {noun}.
                        "{noun} has no text or img. Please make sure all "
                        "{noun_lower}s have one or the other, or both."
                    ).format(noun=noun, noun_lower=noun.lower())
                    result['errors'].append(error_message)
```

The same comment applies to the next `error_message` in L182 below.